src/compiler/scala/tools/nsc/typechecker/NamesDefaults.scala

```diff
@@ -29,6 +29,12 @@ trait NamesDefaults { self: Analyzer =>
   // as an attachment in the companion module symbol
   class ConstructorDefaultsAttachment(val classWithDefault: ClassDef, var companionModuleClassNamer: Namer)
 
+  // Attached to the synthetic companion `apply` method symbol generated for case classes, holds
+  // the set contains all default getters for that method. If the synthetic `apply` is unlinked in
+  // its completer because there's a user-defined matching method (PR #5730), we have to unlink the
+  // default getters as well.
+  class CaseApplyDefaultGetters(val defaultGetters: mutable.Set[Symbol] = mutable.Set.empty)
```

Comment:
Add a quick note on the (current) lifetime of the attachment? Similar to my comment at `sym.removeAttachment`